### ISSUE TITLE
Widget in AJAX mode no long shows current show On-Air

### ISSUE THREAD

**Issue by andyb2000** [netmix/radio-station#477]
**Describe the bug**
Previously I had a side and footer widget that used the "Current Show On-Air" widget to display current presenter, image, etc. After recent upgrade (v2.5.6) this has stopped working, on the website I now see the placeholder when no show is scheduled. If I switch AJAX to Off in the widget settings, it correctly displays the current presenter.

![image](https://github.com/netmix/radio-station/assets/1208595/92f89ffb-d8ab-4b3b-a7cb-987e1d101663)

(Note, here I have set AJAX to Off to allow the info to be shown on the website)

**To Reproduce**
1. Add schedule
2. enable widget and set AJAX to on
3. go on website, it will display the "No Current Show Text:" no matter what time.

**Expected behavior**
The current show displayed in the widget

**Desktop (please complete the following information):**
 - OS: Tested on various (Ubuntu, Windows)
 - Browser: Tested on multiple: Chrome, Edge
 - Version: various

**Smartphone (please complete the following information):**
 - Device: Tested on various (Iphone and android)
 - OS: all
 - Browser all
 - Version all

**Additional context**

In the browser debug I am seeing a warning (not a js error) about quirk mode being activated for this widget plugin:

```
One or more documents in this page is in Quirks Mode, which will render the affected document(s) with quirks incompatible with the current HTML and CSS specifications.
Quirks Mode exists mostly due to historical reasons. If this is not intentional, you can add or modify the DOCTYPE to be `<!DOCTYPE html>` to render the page in No Quirks Mode.
https://mywebsite.co.uk/wp-admin/admin-ajax.php?action=radio_station_current_show&instance=1&timestamp=1701248591&title=Currently%20On-Air&ajax=on&dynamic=0&no_shows=The%20Best%20Music&hide_empty=0&show_link=1&title_position=above&show_avatar=1&avatar_width=&avatar_size=thumbnail&show_sched=1&show_all_sched=0&countdown=0&time_format=12&display_hosts=1&link_hosts=0&show_desc=0&show_playlist=0&show_encore=1&widget=1&block=0&id=1&for_time=0

``` 

If I call the above php/url direct to view output, it also only shows the default placeholder and not the current show so I suspect it's something in the parameters going wrong?

**Comment by novesia** [comment]
Auch bei mir  ist das Problem aufgetreten

**Comment by andyb2000** [comment]
Confirmed, applied update and working great.
Thank you so much guys :)

**Comment by novesia** [comment]
still have problems. It is displayed but not updated. Current programs and subsequent programs are not updated. This only happens when I empty the cache in the dashboard/Wordpress.

 

Gesendet: Donnerstag, 07. Dezember 2023 um 03:46 Uhr
Von: "majick777" ***@***.***>
An: "netmix/radio-station" ***@***.***>
Cc: "novesia" ***@***.***>, "Mention" ***@***.***>
Betreff: Re: [netmix/radio-station] Widget in AJAX mode no long shows current show On-Air (Issue #477)


 

@andyb2000 @novesia
This issue is resolved in the latest release (2.5.7) now available via Github and the WordPress.org Repository.
Try that and let us know if you have any further issues with this and we can take a look.

**Comment by tonyzeoli** [comment]
Can you check the widget setting to enable AJAX after the update? It's just a suggestion. There could still be a bug but thought I'd mention it in case it got turned off. We'll also wait to see what Tony Hayes adds when he sees your message.
